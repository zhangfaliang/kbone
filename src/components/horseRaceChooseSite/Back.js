import React from 'react';
import IconFont from '../iconFont';
import styles from './index.less';

const Back = ({
  prefixCls = 'default',
  onClickBack = () => {
    console.log('onClickBack');
    return 'onClickBack';
  },
}) => {
  return (
    <div onClick={onClickBack} className={styles[`${prefixCls}-back`]}>
      <IconFont type="mycopy" />
    </div>
  );
};
Back.COMPONENT_NAME = 'BACK';
export default Back;
