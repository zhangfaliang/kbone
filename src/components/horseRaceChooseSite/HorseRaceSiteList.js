import React from 'react';
import classnames from 'classnames';
import styles from './index.less';
import { get } from 'lodash';
import { offset } from '../../utils/commonFn.js';

class HorseRaceSiteList extends React.Component {
  static COMPONENT_NAME = 'HORSERACESITELIST';
  constructor(props) {
    super(props);
    this.state = { activeKey: props.activeKey };
    this.maxLength = 8;
    this.siteWrap = React.createRef();
  }

  static getDerivedStateFromProps(props, state) {
    if (props.activeKey !== state.activeKey) {
      return { activeKey: props.activeKey };
    }
    return null;
  }
  setHeight = () => {
    const siteWrap = this.siteWrap.current;
    const bodyHeight = document.querySelector('body').clientHeight;
    const maxHeight = bodyHeight - offset(siteWrap).top;
    if (maxHeight < siteWrap.clientHeight) {
      siteWrap.style.height = `${maxHeight}px`;
      siteWrap.style.overflow = 'scroll';
    }
  };

  scrollTopFn = isDidMount => {
    const { sites } = this.props;
    const siteWrap = this.siteWrap.current;
    let activeIndex, minChildrenHeight;
    for (let i = 0; i < sites.length; i++) {
      const site = sites[i];
      if (site.key == this.props.activeKey) {
        activeIndex = i;
        break;
      }
    }
    minChildrenHeight = get(siteWrap, 'children.0.clientHeight', 0);
    const maxChildrenHeight = minChildrenHeight * (activeIndex + 1);
    if (maxChildrenHeight > siteWrap.clientHeight) {
      siteWrap.scrollTop = maxChildrenHeight - siteWrap.clientHeight;
    }
  };

  getSnapshotBeforeUpdate(prevProps, prevState) {
    if (this.props.sites.length > this.maxLength) {
      return true;
    }
    return null;
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    if (snapshot) {
      this.setHeight();
      this.scrollTopFn();
    }
  }

  componentDidMount() {
    this.setHeight();
    this.scrollTopFn();
  }

  handleItem = site => {
    const { handleItem } = this.props;
    this.setState({ activeKey: site.key });
    handleItem && handleItem(site.value, site.key);
  };
  render() {
    const { sites, prefixCls } = this.props;
    return (
      <div ref={this.siteWrap} className={styles[`${prefixCls}-site-block`]}>
        {sites.map(site => {
          const clsStr = classnames({
            [styles.site]: true,
            [styles.active]: this.state.activeKey == site.key,
          });
          return (
            <span
              onClick={() => {
                this.handleItem(site);
              }}
              className={clsStr}
              key={site.key}
            >
              {site.value}
            </span>
          );
        })}
      </div>
    );
  }
}
HorseRaceSiteList.defaultProps = {
  prefixCls: 'default',
  activeKey: '',
  sites: [],
  handleItem: key => {
    console.log(key);
    return 'handleItem';
  },
};

export default HorseRaceSiteList;
