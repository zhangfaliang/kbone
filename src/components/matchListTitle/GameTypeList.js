import React from 'react';
import styles from './index.less';

const GameTypeList = ({
  footerText = '',
  gamePlays = [],
  prefixCls = 'default',
  handleItem = (item, gamePlayKey) => {
    console.log(item, gamePlayKey);
  },
}) => {
  const clsName = `${prefixCls || 'default'}-game-type-list`;
  return (
    <div
      className={styles[clsName]}
      onClick={e => {
        e.stopPropagation();
      }}
    >
      {Object.keys(gamePlays).map(gamePlayKey => {
        const item = gamePlays[gamePlayKey];
        return (
          <span
            key={item}
            onClick={e => {
              e.stopPropagation();
              handleItem(item, gamePlayKey);
            }}
            className={styles.item}
          >
            {item}
          </span>
        );
      })}

      <div
        onClick={e => {
          e.stopPropagation();
        }}
        className={styles.footer}
      >
        {footerText}
      </div>
    </div>
  );
};
GameTypeList.COMPONENT_NAME = 'GAMETYPELIST';
export default GameTypeList;
