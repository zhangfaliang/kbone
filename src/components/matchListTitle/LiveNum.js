import React, { Fragment } from 'react';
import IconFont from '../iconFont';
import styles from './index.less';

const LiveNum = ({
  number = '',
  isShow = false,
  prefixCls = 'default',
  handleLive = () => {
    console.log('handleLiveNum');
  },
}) => {
  const clsNameStr = `${prefixCls || 'default'}-live-num`;
  const handleLiveNum = e => {
    e.preventDefault();
    handleLive && handleLive();
  };
  return (
    <Fragment>
      {isShow && (
        <div onClick={handleLiveNum} className={styles[clsNameStr]}>
          <span className={styles.liveImg} />
          <span className={styles.num}>{number}</span>
          <span className={styles.icon}>
            <IconFont type="mycopy" />
          </span>
        </div>
      )}
    </Fragment>
  );
};
LiveNum.COMPONENT_NAME = 'LIVENUM';
export default LiveNum;
