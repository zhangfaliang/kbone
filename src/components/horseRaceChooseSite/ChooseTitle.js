import React from 'react';
import Back from './Back';
import { ChoosePlay } from '../matchListTitle';
import styles from './index.less';

class ChooseTitle extends React.Component {
  static COMPONENT_NAME = 'CHOOSETITLE';
  onClickBack = () => {
    const { onClickBack } = this.props;
    onClickBack && onClickBack();
  };
  handleChoose = activeFlag => {
    const { handleChoose } = this.props;
    handleChoose && handleChoose(activeFlag);
  };
  render() {
    const { children, chooseText, choosePlayActive, prefixCls } = this.props;
    const childrens = React.Children.map(children, option => {
      const { onClickBack, ...other } = option.props;
      if (option.type.COMPONENT_NAME === 'BACK') {
        return <Back onClickBack={this.onClickBack} {...other} />;
      } else if (option.type.COMPONENT_NAME === 'CHOOSEPLAY') {
        return (
          <ChoosePlay
            {...other}
            active={choosePlayActive}
            chooseText={chooseText}
            handleChoose={this.handleChoose}
          />
        );
      }
    });
    return <div className={styles[`${prefixCls}-choose-title`]}>{childrens}</div>;
  }
}
ChooseTitle.defaultProps = {
  prefixCls: 'default',
  onClickBack: () => {
    console.log('onClickBack');
    return 'onClickBack';
  },
  choosePlayActive: false,
  chooseText: '',
  handleChoose: activeFlag => {
    console.log('handleChoose', activeFlag);
    return 'handleChoose';
  },
};

export default ChooseTitle;
