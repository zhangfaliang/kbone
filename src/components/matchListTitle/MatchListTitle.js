import React, { Component } from 'react';
import LiveNum from './LiveNum';
import ChoosePlay from './ChoosePlay';
import LeftTitle from '../label/LeftTitle';
import LabelRight from '../label/LabelRight';
import GamePlayModule from './GamePlayModule';
import { offset } from '../../utils/commonFn.js';
import styles from './index.less';

class MatchListTitle extends Component {
  constructor(props) {
    super(props);
    this.listTitle = React.createRef();
    this.state = {
      isOpenFlag: false,
      selectItem: this.props.chooseText,
      offsetTop: 0,
    };
  }

  shouldComponentUpdate(nextProps) {
    if (nextProps.chooseText !== this.props.chooseText) {
      this.setState({
        selectItem: nextProps.chooseText,
      });
    }
    return true;
  }
  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.choosePlayOpenFlag !== prevState.isOpenFlag) {
      return {
        isOpenFlag: nextProps.choosePlayOpenFlag,
      };
    }
    return null;
  }

  handleChoose = isOpenFlag => {
    const { handleChoose } = this.props;
    this.setState({
      isOpenFlag,
    });
    handleChoose && handleChoose(isOpenFlag);
  };
  handleItem = (item, gamePlayKey) => {
    const { handleItem } = this.props;
    this.setState({
      selectItem: item,
    });
    this.setState({
      isOpenFlag: false,
    });
    handleItem && handleItem(item, gamePlayKey);
  };
  componentDidMount() {
    const offsetTop = offset(this.listTitle.current).top + this.listTitle.current.clientHeight;
    this.setState({ offsetTop: offsetTop });
  }

  render() {
    const { prefixCls, children } = this.props;
    const { isOpenFlag, selectItem, offsetTop } = this.state;

    const childrens = React.Children.map(children, option => {
      const { isOpen, chooseText, ...other } = option.props;
      if (option.type.COMPONENT_NAME === 'LIVENUM') {
        return <LiveNum {...other} handleLive={this.props.handleLive} />;
      } else if (option.type.COMPONENT_NAME === 'CHOOSEPLAY') {
        return (
          <ChoosePlay
            {...other}
            chooseText={selectItem}
            handleChoose={this.handleChoose}
            active={isOpenFlag}
          />
        );
      } else if (option.type.COMPONENT_NAME === 'LEFTNODE') {
        return <LeftTitle {...other} clickLeftTitleText={this.props.clickLeftTitleText} />;
      } else if (option.type.COMPONENT_NAME === 'GAMEPLAYMODULE') {
        return (
          <GamePlayModule
            isOpen={isOpenFlag}
            {...other}
            handleItem={this.handleItem}
            handleLayer={this.handleChoose}
            offsetTop={offsetTop}
          />
        );
      } else if (option.type.COMPONENT_NAME === 'LABELRIGHT') {
        return <LabelRight {...other} />;
      }
    });
    const clsName = `${prefixCls || 'default'}-match-list-title`;
    return (
      <div ref={this.listTitle} className={styles[clsName]}>
        {childrens}
      </div>
    );
  }
}
MatchListTitle.defaultProps = {
  prefixCls: 'default',
  chooseText: '',
  handleLive: () => {
    console.log('handleLive --title ');
    return 'handleLive';
  },
  handleChoose: () => {
    console.log('handleChoose --title ');
    return 'handleChoose';
  },
  clickLeftTitleText: () => {
    console.log('clickLeftTitleText --title ');
    return 'clickLeftTitleText';
  },
  handleItem: item => {
    console.log(item + 'handleItem --title ');
    return 'handleItem';
  },
};
export default MatchListTitle;
