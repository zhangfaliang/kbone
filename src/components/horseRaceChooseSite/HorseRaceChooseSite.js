import React from 'react';
import ChooseTitle from './ChooseTitle';
import { GamePlayModule } from '../matchListTitle';
import styles from './index.less';

class HorseRaceChooseSite extends React.Component {
  onClickBack = () => {
    const { onClickBack } = this.props;
    onClickBack && onClickBack();
  };
  handleChoose = closeFlag => {
    const { handleChooseSite } = this.props;
    handleChooseSite && handleChooseSite(closeFlag);
  };
  handleSiteLayer = closeFlag => {
    const { handleSiteLayer } = this.props;
    handleSiteLayer && handleSiteLayer(closeFlag);
  };
  handleSiteItem = (item, siteKey) => {
    const { handleSiteItem } = this.props;
    handleSiteItem && handleSiteItem(item, siteKey);
  };
  render() {
    const { children, prefixCls } = this.props;
    const childrens = React.Children.map(children, option => {
      const { ...other } = option.props;
      if (option.type.COMPONENT_NAME === 'CHOOSETITLE') {
        return (
          <ChooseTitle {...other} onClickBack={this.onClickBack} handleChoose={this.handleChoose} />
        );
      } else if (option.type.COMPONENT_NAME === 'GAMEPLAYMODULE') {
        return (
          <GamePlayModule
            {...other}
            handleItem={this.handleSiteItem}
            handleLayer={this.handleSiteLayer}
          />
        );
      }
    });
    return <div className={styles[`${prefixCls}-choose-site`]}>{childrens}</div>;
  }
}
HorseRaceChooseSite.defaultProps = {
  prefixCls: 'default',
  handleSiteLayer: () => {
    console.log('handleLayer');
    return 'handleLayer';
  },
  handleSiteItem: (item, siteKey) => {
    console.log(item, siteKey);
    return 'handleItem';
  },
  onClickBack: () => {
    console.log('onClickBack');
  },
  handleChooseSite: flg => {
    console.log('flg');
    return 'handleChooseSite';
  },
};
export default HorseRaceChooseSite;
