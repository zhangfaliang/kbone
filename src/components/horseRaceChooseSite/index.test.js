import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import ChooseTitle from './ChooseTitle';
import Back from './Back';
import HorseRaceChooseSite from './HorseRaceChooseSite';
import HorseRaceSiteList from './HorseRaceSiteList';
import { ChoosePlay } from '../matchListTitle';
import { GamePlayModule } from '../matchListTitle';
import HorseRaceNum from '../horseRaceNum';

describe('HorseRaceBetButtonCell', async () => {
  it('Back', () => {
    expect.assertions(1);
    const wrapper = render(<Back />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('click Back', () => {
    expect.assertions(1);
    const wrapper = mount(<Back />);
    const selectBetBtn = wrapper.prop('onClickBack');
    const click = jest.fn(selectBetBtn);
    wrapper.simulate('click');
    expect(click()).toBe(undefined);
  });

  it('ChooseTitle', () => {
    expect.assertions(1);
    const wrapper = render(<ChooseTitle />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('ChooseTitle ChoosePlay', () => {
    expect.assertions(1);
    const wrapper = mount(
      <ChooseTitle>
        <Back />
        <ChoosePlay />
      </ChooseTitle>
    );

    const selectChoose = wrapper.prop('handleChoose');
    const handleChoose = jest.fn(selectChoose);
    wrapper.find('ChoosePlay').simulate('click');
    expect(handleChoose()).toBe('handleChoose');
  });
  it('ChooseTitle onClickBack', () => {
    expect.assertions(1);
    const wrapper = mount(
      <ChooseTitle>
        <Back />
        <GamePlayModule />
      </ChooseTitle>
    );
    const selectBetBtn = wrapper.prop('onClickBack');
    const click = jest.fn(selectBetBtn);
    wrapper.find('Back').simulate('click');
    expect(click()).toBe('onClickBack');
  });

  it('HorseRaceChooseSite', () => {
    expect.assertions(1);
    const wrapper = render(
      <HorseRaceChooseSite>
        <ChooseTitle />
        <GamePlayModule />
      </HorseRaceChooseSite>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseRaceSiteList handleSiteItem', () => {
    expect.assertions(1);
    const wrapper = render(<HorseRaceSiteList sites={[{ key: '43534', value: 'dfdsfsdf' }]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseRaceChooseSite handleSiteLayer', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceChooseSite>
        <ChooseTitle />
        <GamePlayModule />
      </HorseRaceChooseSite>
    );
    const selectBetBtn = wrapper.prop('handleSiteLayer');
    const click = jest.fn(selectBetBtn);
    wrapper.find('GamePlayModule').simulate('click');
    expect(click()).toBe('handleLayer');
  });
  it('HorseRaceChooseSite handleChooseSite', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceChooseSite>
        <ChooseTitle />
        <GamePlayModule />
      </HorseRaceChooseSite>
    );
    const selectBetBtn = wrapper.prop('handleChooseSite');
    const click = jest.fn(selectBetBtn);
    wrapper.find('GamePlayModule').simulate('click');
    expect(click()).toBe('handleChooseSite');
  });
  it('HorseRaceChooseSite handleSiteItem', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceChooseSite>
        <ChooseTitle />
        <GamePlayModule>
          <HorseRaceSiteList sites={[{ key: '43534', value: 'dfdsfsdf' }]} />
        </GamePlayModule>
      </HorseRaceChooseSite>
    );
    const selectBetBtn = wrapper.prop('handleSiteItem');
    const click = jest.fn(selectBetBtn);
    wrapper
      .find('GamePlayModule')
      .find('HorseRaceSiteList')
      .find('span')
      .simulate('click');
    expect(click()).toBe('handleItem');
  });

  it('HorseRaceSiteList handleSiteItem', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceSiteList
        maxHeight={2}
        activeKey="435341"
        sites={[
          { key: '435341', value: 'dfdsfsdf' },
          { key: '435342', value: 'dfdsfsdf' },
          { key: '435343', value: 'dfdsfsdf' },
          { key: '435344', value: 'dfdsfsdf' },
          { key: '435345', value: 'dfdsfsdf' },
          { key: '435346', value: 'dfdsfsdf' },
          { key: '435347', value: 'dfdsfsdf' },
          { key: '435348', value: 'dfdsfsdf' },
          { key: '435349', value: 'dfdsfsdf' },
          { key: '4353410', value: 'dfdsfsdf' },
          { key: '4353411', value: 'dfdsfsdf' },
          { key: '4353412', value: 'dfdsfsdf' },
          { key: '4353413', value: 'dfdsfsdf' },
          { key: '4353414', value: 'dfdsfsdf' },
          { key: '4353415', value: 'dfdsfsdf' },
          { key: '4353416', value: 'dfdsfsdf' },
        ]}
      />
    );
    wrapper.setProps({
      activeKey: '4353416',
    });
    const selectBetBtn = wrapper.prop('handleItem');
    const click = jest.fn(selectBetBtn);
    wrapper.simulate('click');
    expect(click()).toBe('handleItem');
  });

  it('HorseRaceSiteList handleSiteItem', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceSiteList
        maxHeight={12}
        activeKey="4353416"
        sites={[
          { key: '435341', value: 'dfdsfsdf' },
          { key: '435342', value: 'dfdsfsdf' },
          { key: '435343', value: 'dfdsfsdf' },
          { key: '435344', value: 'dfdsfsdf' },
          { key: '435345', value: 'dfdsfsdf' },
        ]}
      />
    );
    wrapper.setProps({
      activeKey: '4353416',
    });
    const selectBetBtn = wrapper.prop('handleItem');
    const click = jest.fn(selectBetBtn);
    wrapper.simulate('click');
    expect(click()).toBe('handleItem');
  });
  it('HorseRaceChooseSite handleSiteItem', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceChooseSite>
        <ChooseTitle>
          <Back />
          <ChoosePlay />
        </ChooseTitle>
      </HorseRaceChooseSite>
    );
    const selectBetBtn = wrapper.prop('handleSiteItem');
    const click = jest.fn(selectBetBtn);
    wrapper
      .find('ChooseTitle')
      .find('ChoosePlay')
      .simulate('click');
    expect(click()).toBe('handleItem');
  });
  it('HorseRaceChooseSite onClickBack', () => {
    expect.assertions(1);
    const wrapper = mount(
      <HorseRaceChooseSite>
        <ChooseTitle>
          <Back />
          <GamePlayModule />
        </ChooseTitle>
      </HorseRaceChooseSite>
    );
    const selectBetBtn = wrapper.prop('onClickBack');
    const click = jest.fn(selectBetBtn);

    wrapper
      .find('ChooseTitle')
      .find('Back')
      .simulate('click');
    expect(click()).toBe(undefined);
  });
});
