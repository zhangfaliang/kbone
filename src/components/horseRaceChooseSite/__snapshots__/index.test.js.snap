// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorseRaceBetButtonCell Back 1`] = `
<div
  class="default-back"
>
  <span
    class="iconfont icon-mycopy"
  />
</div>
`;

exports[`HorseRaceBetButtonCell ChooseTitle 1`] = `
<div
  class="default-choose-title"
/>
`;

exports[`HorseRaceBetButtonCell HorseRaceChooseSite 1`] = `
<div
  class="default-choose-site"
>
  <div
    class="default-choose-title"
  />
  <div
    class="default-game-play-module open"
    style="top:undefinedpx"
  />
</div>
`;

exports[`HorseRaceBetButtonCell HorseRaceSiteList handleSiteItem 1`] = `
<div
  class="default-site-block"
>
  <span
    class="site"
  >
    dfdsfsdf
  </span>
</div>
`;
